scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+}
+
+# The following function:
+# 1. checks if CLONE plugin is loaded. If not loads it.
+# 2. checks if SSL cert and key are configured for any of and in that order:
+#    a. CLONE plugin
+#    b. MySQL in general
+#    c. in [sst] section of my.cnf
+# 3. If SSL is configured, but not for the CLONE plugin explicitly, sets up
+#    corresponging CLONE plugin variables
+#
+# Requires environment variables: MYSQL_ACLIENT, MYSQL_PWD
+# Sets the following environment variables:
+# CLEANUP_PLUGIN, REQUIRE_SSL, CLONE_SSL_CERT,
+# CLONE_SSL_KEY, CLONE_SSL_CA, CLIENT_SSL_OPTIONS, CLEANUP_SSL
+#
+setup_clone_plugin()
+{
+    
+    # Either donor or recipient
+    local -r ROLE=$1
+    
+    wsrep_log_debug "-> ############## SSL SECTION [START] ($ROLE)############"
+
+    local CLONE_PLUGIN_LOADED=`$MYSQL_ACLIENT -e "SELECT COUNT(*) FROM INFORMATION_SCHEMA.PLUGINS WHERE PLUGIN_TYPE = 'CLONE';"`
+    
+    wsrep_log_debug "->CLONE_PLUGIN_LOADED: $CLONE_PLUGIN_LOADED"
+    
+    if [ "$CLONE_PLUGIN_LOADED" -eq 0 ]
+    then
+        wsrep_log_info "Installing CLONE plugin"
+
+        # INSTALL PLUGIN is replicated by default, so we need to switch off
+        # session replication on donor
+        if [ "$ROLE" = "donor" ]
+        then
+            WSREP_OFF="SET SESSION wsrep_on=OFF; "
+        else
+            WSREP_OFF="" # joiner does not have replication enabled
+        fi
+        $MYSQL_ACLIENT -e "${WSREP_OFF}INSTALL PLUGIN CLONE SONAME 'mysql_clone.so';"
+        if [ "$ROLE" == "donor" ]; then
+            CLEANUP_CLONE_PLUGIN="yes"
+        fi
+        CLONE_SSL_CERT="NULL"
+        CLONE_SSL_KEY="NULL"
+        CLONE_SSL_CA="NULL"
+    else
+        CLEANUP_CLONE_PLUGIN="no"
+        CLONE_SSL_CERT=`$MYSQL_ACLIENT -e "SELECT @@clone_ssl_cert"`
+        CLONE_SSL_KEY=`$MYSQL_ACLIENT -e "SELECT @@clone_ssl_key"`
+        CLONE_SSL_CA=`$MYSQL_ACLIENT -e "SELECT @@clone_ssl_ca"`
+    fi
+
+    wsrep_log_debug "-> CLONE_SSL_CERT: $CLONE_SSL_CERT; CLONE_SSL_KEY: $CLONE_SSL_KEY; CLONE_SSL_CA: $CLONE_SSL_CA"
+
+    local CLIENT_SSL_CERT=$(parse_cnf sst ssl_cert "")
+    local CLIENT_SSL_KEY=$(parse_cnf sst ssl_key "")
+    local CLIENT_SSL_CA=$(parse_cnf sst ssl_ca "")
+    local CLIENT_SSL_MODE=$(parse_cnf sst ssl_mode "")
+
+    if [ -z "$CLIENT_SSL_CERT" -o "$CLIENT_SSL_CERT" == "" ]
+    then
+        CLIENT_SSL_CERT=$(parse_cnf client ssl_cert "")
+        CLIENT_SSL_KEY=$(parse_cnf client ssl_key "")
+        CLIENT_SSL_CA=$(parse_cnf client ssl_ca "")
+        CLIENT_SSL_MODE=$(parse_cnf client ssl_mode "")
+    fi
+
+    local SERVER_SSL_CERT=`$MYSQL_ACLIENT -e "SELECT @@ssl_cert"`
+    local SERVER_SSL_KEY=`$MYSQL_ACLIENT -e "SELECT @@ssl_key"`
+    local SERVER_SSL_CA=`$MYSQL_ACLIENT -e "SELECT @@ssl_ca"`
+
+    [ "$SERVER_SSL_CERT" = "NULL" ] && SERVER_SSL_CERT=
+    [ "$SERVER_SSL_KEY" = "NULL" ] && SERVER_SSL_KEY=
+    [ "$SERVER_SSL_CA" = "NULL" ] && SERVER_SSL_CA=
+
+    if [ "$CLONE_SSL_CERT" == "NULL" -o "$CLONE_SSL_CERT" = "" ]
```

Comment:
Please address Kamil's comments
https://github.com/percona/percona-xtradb-cluster/pull/2018/files#r1939078846
>Mixture of single `=` and double `==` in one if

Reply:
If I recall correctly we agreed to do this later, when we will beautify the script a bit more. there are many other things that can be changed but not affecting the functionalities.
If we decide to change all now fine for me it will just take other time.

Reply:
Let's address it in a separate PR